packages/rich-text/src/helpers/editor.ts

```diff
@@ -27,6 +28,20 @@ export function hasSelectionText(editor) {
     : false;
 }
 
+type NodeEntry = [CustomElement, Path];
+export function getNodeEntryFromSelection(
+  editor: Editor | SPEditor,
+  nodeType: BLOCKS | INLINES
+): NodeEntry | [] {
+  if (!editor.selection) return [];
+  const { path } = editor.selection.focus;
+  for (let i = 0; i < path.length; i++) {
+    const nodeEntry = Editor.node(editor, path.slice(0, i + 1)) as NodeEntry;
+    if (nodeEntry[0].type === nodeType) return nodeEntry;
+  }
+  return []
+}
```

Comment:
This is a slight perf bump over udecode's `getNodes` that passes us the `[Node, Path]` tuple direct from Slate's `Editor` interface, which we need as often as not. I put it in `helpers` because we use it for a slight improvement to HRs later in this file.